packages/contracts-bedrock/test/kontrol/deployment/DeploymentSummary.t.sol

```diff
@@ -73,3 +76,42 @@ contract DeploymentSummary_Test is DeploymentSummary, OptimismPortal_Test {
     ///         the L2OutputOracle, which is needed in this test
     function test_isOutputFinalized_succeeds() external override { }
 }
+
+contract DeploymentSummary_TestL1CrossDomainMessenger is DeploymentSummary, L1CrossDomainMessenger_Test {
+    /// @notice super.setUp is not called on purpose
+    function setUp() public override {
+        // Recreate Deployment Summary state changes
+        DeploymentSummary deploymentSummary = new DeploymentSummary();
+        deploymentSummary.recreateDeployment();
+
+        // Set summary addresses
+        optimismPortal = OptimismPortal(payable(OptimismPortalProxyAddress));
+        superchainConfig = SuperchainConfig(SuperchainConfigProxyAddress);
+        l2OutputOracle = L2OutputOracle(L2OutputOracleProxyAddress);
+        systemConfig = SystemConfig(SystemConfigProxyAddress);
+        l1CrossDomainMessenger = L1CrossDomainMessenger(L1CrossDomainMessengerProxyAddress);
+
+        // Set up utilized addresses
+        alice = makeAddr("alice");
+        bob = makeAddr("bob");
+        vm.deal(alice, 10000 ether);
+        vm.deal(bob, 10000 ether);
+    }
+
+    /// @dev Skips the first line of `super.test_constructor_succeeds` because
+    ///      we're not exercising the `Deploy` logic in these tests. However,
+    ///      the remaining assertions of the test are important to check
+    function test_constructor_succeeds() external override {
```

Comment:
Curious if you've looked into the `ChainAssertions.sol` lib and how that would compare to this approach? I'm ok with deferring that to an improvement for a follow up PR

Reply:
Not in-depth, but it doesn't seem very trivial to exercise with the summarization. I think it's a nice path to explore! But perhaps outside of this PR. Here are some initial thoughts
- It looks to me that the `ChainAssertions.sol` as-is checks that the files produced by the `Deploy.s.sol` script are correct, and also, this logic is exercised when running our custom `KontrolDeployment` script, since its using the same functions (e.g., [this one](https://github.com/runtimeverification/_audits_Ethereum-optimism_pausability/blob/1a94b1dc20c18245217cd64173ff74aea32acc45/packages/contracts-bedrock/scripts/Deploy.s.sol#L967)). It is true that we're weakening the logic of what's being tested by replacing `mustGetAddress` by `getAddress` when running the script tho
- If what I say above is correct (and apologies if it's not), a nice path forward would be to exercise the same checks as in `ChainAssertions` but with the `DeploymentSummary` directly, or the `Kontrol-Deploy.json`
What I'm trying to say is that what seems to not test atm is the `stateDiff` modifier nor Konrol's process for creating the summaries. Please let me know if any of this is incorrect

Reply:
Thanks for looking into it. I linked to this comment in our shared google doc so we can revisit and unblock this PR